src/fileioc/fileioc.asm

```diff
@@ -202,15 +202,15 @@ ti_Resize:
 ;  sp + 6 : slot index
 ; return:
 ;  hl = new size if no failure
-	pop	de
-	pop	hl			; hl = new size
+	pop	hl
+	pop	de			; de = new size
 	pop	bc			; c = slot
 	push	bc
-	push	hl
 	push	de
+	push	hl
 	call	util_is_slot_open
```

Comment:
`hl` being destroyed here was the other half of the issue. Now the new size is popped into `de` instead, which is not destroyed.